mesa/space.py

```diff
@@ -666,10 +666,13 @@ def find_neighbors(pos: Coordinate, radius: int) -> None:
             if radius > 1:
                 [find_neighbors(coords, radius - 1) for coords in adjacent]
 
+        explored, coordinates = dict(), set()
         find_neighbors(pos, radius)
 
-        if not include_center and pos in coordinates:
-            coordinates.remove(pos)
+        if include_center:
```

Comment:
Just commenting for other reviewers: this code chunk is a refactor, and has nothing to do with the memoization.

Reply:
yes, you are right the two parts you pointed out are not related to the memoization, but I think they are nonetheless an improvement because you don't need to test the include_center condition inside the recursive function but just at the end, but I think that since this is a different change from the memoization part it's better to address it in another pr

Reply:
The problem with your refactor is that there are holes in the `adjacent` regardless of `include_center` is true or not, because you removed L649-650. If you can't decisively prove that this doesn't change current behavior, you should remove the refactor.

Reply:
we discuss it in another pr, I followed your suggestions to remove refactoring from this one :-)